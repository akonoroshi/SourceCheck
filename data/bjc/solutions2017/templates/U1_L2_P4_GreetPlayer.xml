<project name="U1_L2_P4_GreetPlayer" app="Snap! 4.0, http://snap.berkeley.edu" version="1"><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAALT0lEQVR4Xu2deVAVRx7Hv9yXywpBIeARFQ/U1WiM1x8oaqJ4AWI0EBMSjVu6aqllWeK1ruVmDYoHcQ2JxhsFFFAElcvVrfKoXdRkXV03MboSAwYURVF5cm51b80UvU/l8R68nsFfV1nWg+n5/ebbH77T09Pdz6aurq4OVEgBSQrYEICSlKewXAECkECQqsBzAbxy5Qry8vLA7s42Njaa/3/UqFHo3bu3VCEpuHkKGAGYmZmJPn36mHc2ibUuX76M8ePHS8yAQpujgADgmTNn0L59e8HxLl68iJEjR+Lhw4cvdcJnz56he/fuKCgo4MfdvHkTnTt3tqqDFhYWYujQoeboQHUkKSAAuGHDBoSHh/NULly4gODgYOTn56Njx44IDAzkt+Xo6Gj4+/sjNjYWGRkZGDZsGEpLS8EAZGX06NH8uOLiYvj5+Vn1slJTU7Fo0SKrxqRglikgALh+/XpMnjyZOxhzvrCwMJSVlcHOzg4eHh4oKSnB3bt34eLiwt3t9u3bcHZ2Rnl5Of/MSm1tLYfR0dGRu6E1+5ApKSlYvHixZYpQbasqYASg4oBWzaKJgjEHJACbSEwrnUYAMCYmRnXATp064USkNwK8HFHg/htc9g7WxNOwh+EOBt5JhVNNBdwcbfFk7l/VvJgDLlmyxErSUZimUMAIQMUBq56Wo/3+EJx+60/oMziQNzK7Fbs4OqD4TlGjY9fW1MDdq22TQXz+UDzGlezHLzP/oubCHJAAbHTTSK0gALh27VrVAWufPUXR4bXo/FGMmmArF2fcjOqB1o62ZiXt+MUl1NTU8LrXr19Hjx49zDqPUskrPhB3Pj0pOODSpUstOidVtq4CRgAqDsgAzMzOQ2hoKNatWwfWsIZHZbBbORJv/PkcKr7PR/GW+Y3K1iHuIn9IUQBkwzaWlDZfDUPRjDzBAQlASxS1fl0BwM8++0x1QFQZkJGVy4dlNm/ejAULFuBZ+UMBwIorZ+E+IgI/rwxDuzWHUVXyE6pLf0FV0Y/wWfAlCtdEwO/3yfgh5DV+ZfabL3C3mjBhAjZu3Ihu3bpZdMVtvx6Owum5ggMuX77conNSZesqYASg+hRcZcDREzkcQE9PT9y/fx+Vjx9xAM0tdpvy1arsFty1a1dzT8XreW8Lws+f5AgOSABaJKnVKwsArlmzRnVA25pKHDmWpX5m43kuTk54MqevWUnaBwxGzadxuHHjBq/Pxg/ZQLUl44Q+20fg9sfZggOuXLnSrPyokhwFjABUHFABcNKkSSokr/v4ID7QB60cGn4IiUi/hntPDCpwzfFaznfHKPwUlSU4IAEoByRzowoArl69WnU8+7pqpGUcV1/NsQC1hqfAihGws7ExKV7d+vOwt7fnx1ZXV5tUpzEH+e18BwUfnRAccNWqVY05BR0rWQEjABUHZACmHj3GAayoqIC3tzce3i3mfUDveXH8Cdh36V60GjgGhX+MhPfvNuJeYgwe5e1XL6l23Tk4ODjwz1VVVU1+qe12vYtbHx4XHJAAbHKZm/WEAoCs8ZR3wY42tTh0JANDhgzB8OHD+bgdKg0CgJ22f4dHeQlwfTMI1aVFcPTzR8HCIDXh6s/PwMnJCV26dMG1a9eabBBamafYfvdo/GfaMcEBmYtT0Y8CRgAqDqgAWL8PWF5aAs/Y/82WMaUoALJjDQaDRQ8cz5sc23FvMG5+kCk4IAFoSsto5xgBQNaBVxzQ2Q5ITksX+oC2trbo2bOnSdmzmTPsX3OWN/aNxY3IDMEB2ZM8Ff0oYASg4oAMwKTUIwKAWrusTgnj8GPEUcEBCUCttdLL8xEAZIO4igO6OtjiwKE0YRxQa2tEOu8fj+vvpwsOyN7mUNGPAkYAKg6oAFi/D2jJoHFzwOufOBE/TD0iOCABqB/4WKYCgOxFvuKArZzskZCcoulbcNekEHw/5bDggGxGDxX9KGAEoOKADMB9SYc0DWC35FD8+700wQEJQP3AZ+SAbDKn4oDuLo7YcyBZ033A7gfDcG1yquCAbFY3Ff0oIDggA1BxQAVALfcBA1LC8a/wFMEBCUD9wGfkgGxBj+KArd2csSshUdO34J6pk3F10iHBAdnKPir6UUBwQAag4ngMwJ37DmgawF5p7+GfocnqG5a0tDQQgPqBz8gB2aJuxQE9f+WKb/YkaLoP2PvwFA6gMsTDVsWxxfVU9KOA4IAMQMUBFQC13Afsk/4+/jExUXBAAlA/8Bk54MKFC1XH8/p1K2zbtVfTt+C+RyM4gPUdcNOmTfpqgVc8W8EBGYCK4zEAv965R9MAvpkRiW/H7xcckADUF9ECgPPnz1cdsK2HO+K/2aXpPmC/zA84gPUdMC4uTl8t8IpnawSg4oAKgFruA751/ENcHLtPcEACUF9ECwDOmzdPdTyf11pj67YdmnJANjNbKadPn4YCYH0H3LJli75a4BXP1ghAxfEUALXmgEFBQTh16hS/7b6dFYX8MXsEByQA9UW0AOCcOXNUx/Nt44ktX23nDyFsRVtOTg7Gjh2rqasbmP0xB7C+A27dulVTOVIyL1fACEDF8RiAX8Rv47uf+vr6Yu7cubh161aDerINzl+0YXh2djY6dOiAgIAA9Tznzp3jm16GhIQ0eG7lgMTERERERGBQzif427u7BAckAE2WURMHCgDOnj1bdcB23l44GB+Db+8Y0L9/f7D1Irm5uQ0uLLp69SrYcE6vXr0wc+ZMDm1kZCTYWwq2RLNNmzb8Z+PGjePOxdaNMOjPnj1r0qq5gwcPYurUqVy8wbnTOYD1HTA+Pl4TwlISpilgBKDigAxAj33hOB6wgu8PzQrbeNyUmc1eXl7o168fioqK+J4y6enpfBvfGTNmwM3NDQ8ePICrqyvfNWHQoEFwd3fnt/iGZlwr8CUnJyM0+B0M//tinB+1Q3BAAtC0htfKUQKAu3fv5vu1MMg6vN4W7RIm4svyt9E35LdayZfncf9eCUZ89we42VZxAJU/CgZ8VFSUpnKlZBrRB2SHZmVlgS2/ZOOA/ilTNK/f2RHbuQOyfQfHjBmj+XwpQVEBwQGVX7GFPQMGDDDpdtvQbdMav2c7+i9btozaVocKPBdAHV4HpaxTBQhAnTZcS0mbAGwpLanT6yAAddpwLSVtArCltKROr4MA1GnDtZS0CcCW0pI6vQ4CUKcN11LSJgBbSkvq9DoIQJ02XEtJmwBsKS2p0+sgAF/QcJcuXeKTZ9nUMirNpwAB+BJtw8LC+G/LysoQGxvLN2h3cXFpvtZ4Bc9MAL6k0R8/foxp06bx6V5KYXMPmStGR0eDff0Ym7pGxXwFCMAGtJs+fTqfwf3/hX3xNpvVnZSUZL76VFPcI5r0EBWYNWsWiouL+Q8rKyv519ayJQvKrZn0slwBcsAXaMiWJ5w8eRLsW91XrFhhudJ0hucqQAASGFIVIAClyk/BCUBiQKoCBKBU+Sk4AUgMSFWAAJQqPwUnAIkBqQoQgFLlp+AEIDEgVQECUKr8FJwAJAakKkAASpWfghOAxIBUBQhAqfJTcAKQGJCqAAEoVX4KTgASA1IVIAClyk/BCUBiQKoCBKBU+Sk4AUgMSFWAAJQqPwUnAIkBqQoQgFLlp+AEIDEgVQECUKr8FJwAJAakKkAASpWfghOAxIBUBQhAqfJTcAKQGJCqAAEoVX4KTgASA1IVIAClyk/BCUBiQKoCBKBU+Sk4AUgMSFWAAJQqPwUnAIkBqQoQgFLlp+AEIDEgVQECUKr8FJwAJAakKkAASpWfghOAxIBUBQhAqfJTcAKQGJCqAAEoVX4KTgASA1IVIAClyk/BCUBiQKoCBKBU+Sk4AUgMSFWAAJQqPwUnAIkBqQoQgFLlp+AEIDEgVYH/AldS6iAflezuAAAAAElFTkSuQmCC</thumbnail><stage name="Stage" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Sprite" idx="1" x="0.2702702702702027" y="-0.8108108108093859" heading="90.00000000000023" scale="1" rotation="1" draggable="true" costume="0" color="80,80,80" pen="tip" id="8"><costumes><list id="9"></list></costumes><sounds><list id="10"></list></sounds><variables></variables><blocks></blocks><scripts><script x="118" y="48"><custom-block s="greet player"></custom-block></script></scripts></sprite><watcher var="player list" style="normal" x="10" y="10" color="243,118,29" extX="120" extY="105"/></sprites></stage><hidden></hidden><headers></headers><code></code><blocks><block-definition s="greet player" type="command" category="sensing"><header></header><code></code><inputs></inputs><script><block s="doAsk"><l>What&apos;s your name?</l></block><block s="doIfElse"><block s="reportListContainsItem"><block var="player list"/><block s="getLastAnswer"></block></block><script><block s="doSayFor"><block s="reportJoinWords"><list><l>Nice to see you again, </l><block s="getLastAnswer"></block><l>.</l></list></block><l>2</l></block></script><script><block s="doSayFor"><block s="reportJoinWords"><list><l>Hi, </l><block s="getLastAnswer"></block><l>! Nice to meet you!</l></list></block><l>2</l></block><block s="doAddToList"><block s="getLastAnswer"></block><block var="player list"/></block></script></block></script></block-definition></blocks><variables><variable name="player list"><list id="56"><item><l>Thomas</l></item><item><l>Bob</l></item></list></variable></variables></project>