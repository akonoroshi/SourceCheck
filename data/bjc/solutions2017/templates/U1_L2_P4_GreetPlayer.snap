snapshot {
    @defBlock BODY {
        script {
            doAsk(literal)
            doIfElse(reportListContainsItem(var, getLastAnswer), script {
                doSayFor(reportJoinWords(list(literal, getLastAnswer, literal)), literal)
            }, script {
                doSayFor(reportJoinWords(list(literal, getLastAnswer, literal)), literal)
                doAddToList(getLastAnswer, var)
            })
        }
    }
    @defVar BLOCK {
        stage {
            sprite {
                @if (BLOCK) {
                    script {
                        evaluateCustomBlock
                    }
                } @unless (BLOCK) {
                    @block BODY
                }
            }
        }
        @if (BLOCK) {
            customBlock {
                @block BODY
            }
        }
    }
    var
}